src/io/mod.rs

```diff
@@ -186,6 +194,25 @@ impl<D: DeserializeOwned> DelimFileReader<D> {
     pub fn read(&mut self) -> Option<Result<D>> {
         self.record_iter.next().map(|result| result.map_err(FgError::ConversionError))
     }
+
+    fn validate_header(header: &StringRecord, delimiter: u8) -> Result<()> {
+        let delim = String::from_utf8(vec![delimiter]).unwrap();
+        let found_header_parts: Vec<&str> = header.iter().collect();
```

Comment:
`HashSet` may be more performant for `contains` checks?

Reply:
You're right. I was thinking that it does not matter for "small" n. But better to be safe than sorry.

FWIW this was an interesting benchmark: https://gist.github.com/daboross/976978d8200caf86e02acb6805961195